Introduce DEV modoption category

robertthepie opened the pull request:
Adds a new Developer Setting that when toggled:
Adds a DEV modoption category per GDT request, as well as shows HIDDEN modoptions.

# Testing Done
- [X] New Modoption categories don't show when the new setting isn't toggled, both before toggling and after untoggling
- [X] New Modoption categories do show when the setting is enabled, even if enabled when already in lobby
- [ ] Joining Multiplayer where there are already options toggled
- [ ] The new unhide options toggle setting is saved for the next time the software is opened

# Before
![image](https://github.com/beyond-all-reason/BYAR-Chobby/assets/59885012/5d1b9789-a007-4cda-9cb9-8eb12fad1cb7)

# After
![image](https://github.com/beyond-all-reason/BYAR-Chobby/assets/59885012/620c64fb-9d23-4fb2-ba01-b3b6698b2bcb)
![image](https://github.com/beyond-all-reason/BYAR-Chobby/assets/59885012/ca931693-50a7-46b5-9327-0acb83e1a7f3)
![image](https://github.com/beyond-all-reason/BYAR-Chobby/assets/59885012/2a7e27ef-dbb2-4e83-aace-e70527574f42)

sprunk commented:
so what's the difference between _DEV and _HIDDEN that warrants the different treatment? isn't _DEV already hidden as well

robertthepie commented:
Updated the inital comment to include the new developer setting, new after images, and moved hidden stuff to their respective category only visible when the enw setting is toggled on.
Time constraint means i don't have time to test when joining a pre-existing lobby and some other stuff.

robertthepie commented:
screwed up a merge conflic further, opps?
